nnmnkwii/datasets/jvs.py

```diff
@@ -0,0 +1,205 @@
+        self.labelmap = labelmap
+        self.labels = None
+        self.max_files = max_files
+        # TODO: currently there are many problem in falset, so we do not support falset
+        self._folders = ["parallel100", "nonpara30", "whisper10"]
+        self._textfilename = "transcripts_utf8.txt"
+        self._wavfoldername = "wav24kHz16bit"
+        self.speaker_info = _parse_speaker_info(data_root)
+        # some of the text in nonpara data have no corresponding wav file, so we need to record it
+        self._nonpara_without_wav = self._validate(False)
+
+    def _validate(self, strict=False):
+        # for each speaker, validate the data
+        # TODO: check lab file
+        nonpara_without_wav = {}
+        for _, speaker in enumerate(self.speakers):
+            # 1. check folder
+            speaker_folder = join(self.data_root, speaker)
+            folders = listdir(speaker_folder)
+            nonpara_without_wav[speaker] = {}
+            for folder in self._folders:
+                assert folder in folders,\
+                    "Can not find {} for {} in its directory {}"\
+                    .format(folder, speaker, speaker_folder)
+            # 2. check wav and txt file,
+            for folder in self._folders:
+                speaker_textfile = join(speaker_folder, folder, self._textfilename)
+                assert isfile(speaker_textfile), "File {} doesn't exist".format(speaker_textfile)
+                speaker_wavfolder = join(speaker_folder, folder, self._wavfoldername)
+                assert isdir(speaker_wavfolder), "Directory {} doesn't exist".format(speaker_wavfolder)
+                # nonpara
+                if folder == "nonpara30":
+                    with open(speaker_textfile, encoding="utf8") as file_:
+                        txtfiles = [line.strip().split(":")[0] for line in file_]
+                    wavfiles = listdir(speaker_wavfolder)
+                    for file_ in txtfiles:
+                        if file_ + ".wav" not in wavfiles:
+                            nonpara_without_wav[speaker][file_] = '_'
+                    assert (len(txtfiles) - len(nonpara_without_wav[speaker])) == len(wavfiles)
+                # strict mode, check whether each text has a corresponding wavfile, if not, report it.
+                if strict:
+                    with open(speaker_textfile, encoding='utf8') as file_:
+                        txtlines = [line.strip().split(':')[0] for line in file_]
+                    wavlines = [basename(line) for line in listdir(speaker_wavfolder)]
+                    for line in txtlines:
+                        if line + '.wav' not in wavlines:
+                            print("Wavfile {} of speaker {} in {} doesn't exists."
+                                  .format(line, speaker, speaker_wavfolder))
+        return nonpara_without_wav
+
+    def collect_files(self, is_wav, nonpara=False, whisper=False):
```

Comment:
Well, how about removing `nonpara` and `whisper` in the function argment, and instead adding the argment `categories=["parallel"]` to the `__init__` method?

Then we can do like this:

```py
jvs.WavFileDataSource(DATA_DIR, speakers, categories=["parallel", "nonpara"])
```
```py
jvs.WavFileDataSource(DATA_DIR, speakers, categories=["nonpara"])
```
```py
jvs.WavFileDataSource(DATA_DIR, speakers, categories=["parallel", "nonpara", "whisper"])
```

This is more flexible. 

Also, as the `collect_files` method is assumed to have no arguments (https://github.com/r9y9/nnmnkwii/blob/7f537439e5da0d953836b3ef8f325a241b31e7a4/nnmnkwii/datasets/__init__.py#L111), it is better to move arguments in the constructer anyway.
```
```